synapse/config/_base.py

```diff
@@ -487,6 +488,8 @@ def generate_config(
                 tls_private_key_path=tls_private_key_path,
             ).values()
         )
+        conf = re.sub('\n{2,}', "\n", conf)
```

Comment:
Is the intent to exclude entries where `genereate_config_section` returns an empty string? If so, it might be clearer to add an `if` to the generator expression.

```suggestion
            ).values()
            if conf
        )
```

I say "might" because it's just an Opinion. Feel free to ignore this if you're not convinced.

Reply:
The goal was to remove empty lines in general, which I thought might be a little confusing to newer users? If it's okay I will stick with what's here although I acknowledge it is not graceful :)